go/vt/vtorc/logic/topology_recovery.go

```diff
@@ -783,7 +783,12 @@ func fixReplica(ctx context.Context, analysisEntry *inst.ReplicationAnalysis) (r
 		return true, topologyRecovery, err
 	}
 
-	err = setReplicationSource(ctx, analyzedTablet, primaryTablet, reparentutil.IsReplicaSemiSync(durabilityPolicy, primaryTablet, analyzedTablet))
+	var heartBeatInterval float64
+	if analysisEntry.Analysis == inst.ReplicaMisconfigured {
+		heartBeatInterval = float64(analysisEntry.ReplicaNetTimeout) / 2
+	}
+
```

Comment:
I could use some advice/opinions on this front. I have so far chosen to only specify `heartBeatInterval` in the express case where VTOrc has registered a misconfiguration. 

The advantage of this approach is that if the replica only has its replication stopped, then VTOrc won't make it run a `CHANGE REPLICATION SOURCE` command.

The drawback is that if the replica has no replication  source configured and also heartbeat interval misconfigured, then VTOrc will first register `NotConnectedToPrimary` and it won't change the heartbeat settings. Only after the replication source has been fixed, will it register `ReplicaMisconfigured`. So the whole process of the fix will take 2 fixes.

Another disadvantage is that so far `fixReplica` has been fixing all the problems on the replica by configuring it again without caring for what was the initial problem that triggered this fix. We will be changing this assumption, as the analysis will also impact how the fix runs. Moreover, this adds another layer of complexity in replication analysis detection wherein VTOrc must register misconfiguration issues before registering replication not running issues. Because if the replication is not running because of misconfiguration, we want to fix the misconfiguration and not just restart replication. So we must register the misconfiguration problem before.

Reply:
The alternative to this approach is to set the heartbeat interval value always. This would mean that whenever we try to fix the replica, we run the `CHANGE SOURCE REPLICATION` command always.

Reply:
Another approach that we could take is to add a method to the `analysis` like `HeartbeatMisconfigured` which returns a boolean if the heartbeat settings are incorrect, and we use this method in detecting the misconfiguration and also in the fixReplica phase to only set the heartbeat value if it is misconfigured.

Reply:
I suggest taking the simpler code path, even if that means that the process fix takes 2 steps. That's immaterial IMO.

Reply:
As discussed, we have decided to take the simpler, least risk approach of just fixing everything on the replica when we notice anything is broken